diff --git a/.gitignore b/.gitignore
index eb44ff7f255..8634ff924cf 100644
--- a/.gitignore
+++ b/.gitignore
@@ -75,3 +75,5 @@ stamp-*
 /gmp*
 /isl*
 /gettext*
+
+/build/
diff --git a/bfd/config.bfd b/bfd/config.bfd
index 02a1a68525e..fb44cace29b 100644
--- a/bfd/config.bfd
+++ b/bfd/config.bfd
@@ -620,6 +620,11 @@ case "${targ}" in
     targ_defvec=i386_pe_vec
     targ_selvecs="i386_pe_vec i386_pei_vec"
     ;;
+  i[3-7]86-*-panuti)
+    targ_defvec=i386_elf32_vec
+    targ_selvecs=
+    targ64_selvecs=x86_64_elf64_vec
+    ;;
   i[3-7]86-*-beos*)
     targ_defvec=i386_elf32_vec
     targ_selvecs="i386_pe_vec i386_pei_vec"
diff --git a/config.sub b/config.sub
index 3d35cde174d..a7643d75105 100755
--- a/config.sub
+++ b/config.sub
@@ -2087,6 +2087,7 @@ case $os in
 	| oskit* \
 	| osx* \
 	| palmos* \
+	| panuti* \
 	| phoenix* \
 	| plan9* \
 	| powermax* \
diff --git a/gas/configure.tgt b/gas/configure.tgt
index 7a5645fd9ae..72da09baf12 100644
--- a/gas/configure.tgt
+++ b/gas/configure.tgt
@@ -251,6 +251,7 @@ case ${generic_target} in
   i386-*-gnu*)				fmt=elf em=gnu ;;
   i386-*-msdos*)			fmt=aout ;;
   i386-*-moss*)				fmt=elf ;;
+  i386-*-panuti*)                       fmt=elf ;;
   i386-*-pe)				fmt=coff em=pe ;;
   i386-*-cygwin*)
    case ${cpu} in
diff --git a/ld/Makefile.am b/ld/Makefile.am
index 3ae3d7f14f6..9e5432b4c14 100644
--- a/ld/Makefile.am
+++ b/ld/Makefile.am
@@ -279,6 +279,7 @@ ALL_EMULATION_SOURCES = \
 	eelf_i386_be.c \
 	eelf_i386_fbsd.c \
 	eelf_i386_haiku.c \
+	eelf_i386_panuti.c \
 	eelf_i386_sol2.c \
 	eelf_i386_vxworks.c \
 	eelf_iamcu.c \
diff --git a/ld/Makefile.in b/ld/Makefile.in
index d879f04a17c..afeb64b9be8 100644
--- a/ld/Makefile.in
+++ b/ld/Makefile.in
@@ -794,6 +794,7 @@ ALL_EMULATION_SOURCES = \
 	eelf_i386_be.c \
 	eelf_i386_fbsd.c \
 	eelf_i386_haiku.c \
+	eelf_i386_panuti.c \
 	eelf_i386_sol2.c \
 	eelf_i386_vxworks.c \
 	eelf_iamcu.c \
@@ -1470,6 +1471,7 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_be.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_fbsd.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_haiku.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_panuti.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_sol2.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_vxworks.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_iamcu.Po@am__quote@
diff --git a/ld/configure.tgt b/ld/configure.tgt
index 60b62e69c08..3ce3ba1894f 100644
--- a/ld/configure.tgt
+++ b/ld/configure.tgt
@@ -401,6 +401,9 @@ i[3-7]86-*-moss*)	targ_emul=i386moss
 i[3-7]86-*-winnt*)	targ_emul=i386pe ;
 			targ_extra_ofiles="deffilep.o pdb.o pe-dll.o"
 			;;
+i[3-7]86-*-panuti)	targ_emul=elf_i386_panuti
+			targ_extra_emuls=elf_i386
+			;;
 i[3-7]86-*-pe)		targ_emul=i386pe ;
 			targ_extra_ofiles="deffilep.o pdb.o pe-dll.o"
 			;;
diff --git a/ld/emulparams/elf_i386_panuti.sh b/ld/emulparams/elf_i386_panuti.sh
new file mode 100644
index 00000000000..faa298e0fcb
--- /dev/null
+++ b/ld/emulparams/elf_i386_panuti.sh
@@ -0,0 +1,2 @@
+source_sh ${srcdir}/emulparams/elf_i386.sh
+TEXT_START_ADDR=0x08048000
